platform/darwin/src/NSCoder+MGLAdditions.mm

```diff
@@ -0,0 +1,16 @@
+#import "NSCoder+MGLAdditions.h"
+
+#import "NSArray+MGLAdditions.h"
+
+@implementation NSCoder (MGLAdditions)
+
+- (void)mgl_encodeLocationCoordinates2D:(std::vector<CLLocationCoordinate2D>)coordinates forKey:(NSString *)key {
+    [self encodeObject:[NSArray mgl_coordinatesFromCoordinates:coordinates] forKey:key];
```

Comment:
Yeah, looks like you’re right. Unfortunately we’ll have to stick to an array of dictionaries. If performance does turn out to be a major issue, we can flatten the array, at the cost of triggering another debate about latitude-longitude order versus longitude-latitude order (or is it longitude-latitude order versus latitude-longitude order?).